### PR TITLE
feat(gatsby-source-contentful): Increase Contentful sync by up to 10x

### PR DESCRIPTION
**Pull request by axe312ger**
* increase default pageLimit to 1000
* add backoff logic that retries with a lower pageLimit till we are able to sync data from Contentful

The first commit in here should be replaced by #30257

### CODE REVIEW

File: packages/gatsby-source-contentful/src/fetch.js [hunk 1]
```diff
@@ -22,8 +22,8 @@ module.exports = async function contentfulFetch({
     integration: `gatsby-source-contentful`,
     responseLogger: response => {
       function createMetadataLog(response) {
-        if (process.env.gatsby_log_level === `verbose`) {
-          return ``
+        if (response) {
+          return null
```
Comment (axe312ger) [inline packages/gatsby-source-contentful/src/fetch.js:26]: see https://github.com/gatsbyjs/gatsby/pull/30257

File: packages/gatsby-source-contentful/src/fetch.js [hunk 2]
```diff
@@ -38,7 +38,7 @@ module.exports = async function contentfulFetch({
       }
 
       // Log error and throw it in an extended shape
-      if (response.isAxiosError) {
+      if (response.isAxiosError && response.response) {
```
Comment (axe312ger) [inline packages/gatsby-source-contentful/src/fetch.js:41]: see https://github.com/gatsbyjs/gatsby/pull/30257

File: packages/gatsby-source-contentful/src/fetch.js [hunk 3]
```diff
@@ -196,22 +196,53 @@ ${formatPluginOptionsForCLI(pluginConfig.getOriginalPluginOptions(), errors)}`,
...
+        }
+        const query = syncToken
+          ? { nextSyncToken: syncToken, ...basicSyncConfig }
+          : { initial: true, ...basicSyncConfig }
+        currentSyncData = await client.sync(query)
+        syncSuccess = true
+      } catch (e) {
+        // Back off page limit if responses content length exceeds Contentfuls limits.
+        if (
+          e.responseData.data.message.indexOf(`Response size too big`) !== -1 &&
```
Comment (wardpeet) [inline packages/gatsby-source-contentful/src/fetch.js:225]: Includes is easier to read
```suggestion
          e.responseData.data.message.includes(`Response size too big`) &&
```

File: packages/gatsby-source-contentful/src/fetch.js [hunk 4]
```diff
@@ -196,22 +196,53 @@ ${formatPluginOptionsForCLI(pluginConfig.getOriginalPluginOptions(), errors)}`,
...
+          currentPageLimit > 1
+        ) {
+          lastCurrentPageLimit = currentPageLimit
+          currentPageLimit = Math.floor((currentPageLimit / 3) * 2) || 1
```
Comment (wardpeet) [inline packages/gatsby-source-contentful/src/fetch.js:229]: Could you document why remove 1/3 :) if it's an arbitrary number that's fine but please add a comment that says so.

**Review by wardpeet — COMMENTED** [review]
Small comments

**Comment by axe312ger** [comment]
@wardpeet done

**Review by wardpeet — APPROVED** [review]
Thanks!

**Pull request merged**